main/modules/importer/scrivener/importer.js

```diff
@@ -0,0 +1,764 @@
+        const newRTFContent = getRTFContents(newJson.lines, file, matchedFileId, bookId)
+        Object.assign(fileContents.rtfContents, newRTFContent)
+        if (newRTFContent.lineId) {
+          fileContents.rtfContents.lineId = Array.isArray(newRTFContent.lineId)
+            ? newRTFContent.lineId[0]
+            : newRTFContent.lineId
+        } else if (!newJson.lines.length) {
+          fileContents.rtfContents.lineId = createNewLine(
+            newJson.lines,
+            { position: 0, title: i18n('Main Plot') },
+            bookId
+          )
+        }
+      } else {
+        const newRTFContent = getRTFContents(newJson.lines, file, matchedFileId, bookId, isSection)
+        Object.assign(fileContents.rtfContents, newRTFContent)
+      }
+    }
+  })
+  return fileContents
+}
+
+function getMatchedRelevantFiles(relevantFiles, childId) {
+  return relevantFiles.filter(({ filePath }) => {
+    const isVer_3_UUID = UUIDFolderRegEx.test(filePath)
+    if (isVer_3_UUID) {
+      return filePath.includes(childId)
+    } else {
+      return getVer_2_7_matchedFiles(filePath, childId)
+    }
+  })
+}
+
+function createSlateEditor(value) {
+  return {
+    type: 'paragraph',
+    children: value && value.children ? value.children : [{ text: '' }],
+  }
+}
+
+// Object, Array <[{}]>, int, int, int -> Object { nextLineId }
+function getRTFContents(lines, rtf, matchId, bookId, isSection) {
+  let nextLineId = Array.isArray(matchId) ? matchId[0] : matchId
+  if (!isSection && rtf.contents && rtf.contents.length) {
+    const rtfContents = rtf.contents.flatMap((content) => {
+      if (content.children && content.children.length) {
+        return content.children.flatMap((childContent) => {
+          // checks for plotlines in rtf
+          if (childContent.text) {
+            if (childContent.text.includes('Plotline: ')) {
+              const lineTitle = childContent.text.split('Plotline: ').pop()
+              const existingPlotline = lines.find((line) => line.title == lineTitle)
+              if (!existingPlotline) {
+                createNewLine(
+                  lines,
+                  {
+                    id: nextLineId,
+                    title: lineTitle,
+                    color: nextColor(lines.length),
+                  },
+                  bookId
+                )
+              } else {
+                nextLineId = existingPlotline.id
+              }
+              return { lineId: nextLineId }
+            } else {
+              // return default lineId if no plotline found
+              return { lineId: 1 }
+            }
+          }
+        })
+      }
+    })
+    return { lineId: rtfContents[0] && rtfContents[0].lineId ? rtfContents[0].lineId : 1 }
+  } else if (isSection && rtf.contents && rtf.contents.length) {
+    return rtf.contents
+  }
+}
+
+function generateNotes(currentState, json, bookId, files, isNewFile) {
+  json.map((item) => {
```

Comment:
If you `map` an array for side-effect (i.e. you throw away the result of mapping), then please use `forEach` instead.
(`map` allocates a new array and also implies to the reader that you're interested in the result.)